server/item/recipe/item.go

```diff
@@ -0,0 +1,94 @@
+package recipe
+
+import (
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+)
+
+// InputItem is a recipe item. It is an item stack inherently, but it also has an extra value for if it applies to all types.
+type InputItem struct {
+	item.Stack
+	// Variants is true if the item applies to all of its possible variants. This is so it can be rendered in the recipe book
+	// for alternative types.
+	Variants bool
+}
+
+// inputItem is an item that is inputted to a crafting menu.
+type inputItem struct {
```

Comment:
Not a huge fan of how this and InputItem are named the same. Maybe suffix it with `Data`?